### PR TITLE
Unsafe local should require /unsafe flag on compilation

### PR DESCRIPTION
**Pull request by jcouv**
**Customer scenario**

Use the `unsafe` modifier on a local function. The compiler should require that you compile with the `/unsafe` flag, but currently doesn't.

**Bugs this fixes:**
Fixes https://github.com/dotnet/roslyn/issues/21485

**Risk**
**Performance impact**
Low. Just adding the same check that we apply to the `unsafe` modifier on methods, but on local functions.

**Is this a regression from a previous update?**
No.

**How was the bug found?**
Reported by customer.

:memo: I reached out to the compat council to confirm this compat break is acceptable.
Update: compat council approved the change.

### CODE REVIEW

**Comment by jcouv** [comment]
@agocke @dotnet/roslyn-compiler for review (one-line change, plus tests). Thanks

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs [hunk 1]
```diff
@@ -31,6 +31,106 @@ internal void VerifyDiagnostics(string source, CSharpCompilationOptions options,
...
+            // no need to declare local function `local` or method `B` as unsafe
+            var compWithUnsafe = CreateStandardCompilation(source, options: TestOptions.UnsafeDebugDll);
+            compWithUnsafe.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LocalInUnsafeBlock()
+        {
+            var source = @"
+unsafe struct C
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Is the `unsafe` intentional here?

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Yes. Andy mentioned that `unsafe struct` is a thing and saves you from marking methods as `unsafe`. The test verifies that local functions behave the same as methods.

**Comment by jcouv** [comment]
test windows_release_unit64_prtest please

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: This test has both `unsafe struct` and `unsafe { ... }`.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Ah, that is not intentional. Thanks!

**Pull request merged**